contracts/abstract/AbstractNFTRewardDelegate.sol

```diff
@@ -0,0 +1,391 @@
+  //*********************************************************************//
+
+  /**
+  @notice NFT redemption is not supported.
+   */
+  // solhint-disable-next-line
+  function didRedeem(JBDidRedeemData calldata _data) external override {
+    // not a supported workflow for NFTs
+  }
+
+  //*********************************************************************//
+  // ---------------------------- IERC165 ------------------------------ //
+  //*********************************************************************//
+
+  function supportsInterface(bytes4 _interfaceId)
+    public
+    view
+    override(ERC721Rari, IERC165)
+    returns (bool)
+  {
+    return
+      _interfaceId == type(IJBFundingCycleDataSource).interfaceId ||
+      _interfaceId == type(IJBPayDelegate).interfaceId ||
+      _interfaceId == type(IJBRedemptionDelegate).interfaceId ||
+      super.supportsInterface(_interfaceId); // check with rari-ERC721
+  }
+
+  //*********************************************************************//
+  // ---------------------- ITokenSupplyDetails ------------------------ //
+  //*********************************************************************//
+
+  function totalSupply() public view override returns (uint256) {
+    return _supply;
+  }
+
+  function tokenSupply(uint256 _tokenId) public view override returns (uint256) {
+    return _ownerOf[_tokenId] != address(0) ? 1 : 0;
+  }
+
+  function totalOwnerBalance(address _account) public view override returns (uint256) {
+    if (_account == address(0)) {
+      revert INVALID_ADDRESS();
+    }
+
+    return _balanceOf[_account];
+  }
+
+  function ownerTokenBalance(address _account, uint256 _tokenId)
+    public
+    view
+    override
+    returns (uint256)
+  {
+    return _ownerOf[_tokenId] == _account ? 1 : 0;
+  }
+
+  //*********************************************************************//
+  // ----------------------------- ERC721 ------------------------------ //
+  //*********************************************************************//
+
+  /**
+    @notice Returns the full URI for the asset.
+  */
+  function tokenURI(uint256 tokenId) public view override returns (string memory) {
+    if (_ownerOf[tokenId] == address(0)) {
+      revert INVALID_TOKEN();
+    }
+
+    if (address(_tokenUriResolver) != address(0)) {
+      return _tokenUriResolver.tokenURI(tokenId);
+    }
+
+    return bytes(_baseUri).length > 0 ? string(abi.encodePacked(_baseUri, tokenId.toString())) : '';
+  }
+
+  /**
+    @notice Returns the contract metadata uri.
+  */
+  function contractURI() public view override returns (string memory contractUri) {
+    contractUri = _contractUri;
+  }
+
+  /**
+    @notice
+    Transfer tokens to an account.
+
+    @param _to The destination address.
+    @param _id NFT id to transfer.
+  */
+  function transfer(address _to, uint256 _id) public override {
+    if (!transferrable) {
+      revert NON_TRANSFERRABLE();
+    }
+    transferFrom(msg.sender, _to, _id);
```

Comment:
you test the transferrable twice then (which is another double sread) -> `super.transferFrom(msg.sender, _to_id);`